### ISSUE TITLE
Creating a road makes the simulation loop

### ISSUE THREAD

**Issue by alessiogambi** [se2p/tool-competition-av#9]
Additionally, the simulation goes into an infinite loop.

this is the code to generate the test:

```
        test.append( (10, 10) )
        test.append( (10, self.map_size -10) )
        test.append( (self.map_size -10, self.map_size -10) )
        test.append( (self.map_size -10, 10) )

```

**Comment by alessiogambi** [comment]
D:\tool-competition-av\simulations\beamng_executor\sim_2020-11-23--14-21-05
ERROR:shapely.geos:TopologyException: Input geom 1 is invalid: Ring Self-intersection at or near point 186 190 at 186 190
Traceback (most recent call last):
  File "D:\tool-competition-av\beamng_executor.py", line 93, in _run_simulation
    sim_data_collector.collect_current_data(oob_bb=True)
  File "D:\tool-competition-av\self_driving\simulation_data_collector.py", line 33, in collect_current_data
    is_oob, oob_counter, max_oob_percentage, oob_distance = self.oob_monitor.get_oob_info(oob_bb=oob_bb, wrt=wrt)
  File "D:\tool-competition-av\self_driving\oob_monitor.py", line 21, in get_oob_info
    is_oob = self.is_oob_bb(tolerance=tolerance, wrt=wrt)
  File "D:\tool-competition-av\self_driving\oob_monitor.py", line 62, in is_oob_bb
    return self.oob_percentage(wrt=wrt) > tolerance
  File "D:\tool-competition-av\self_driving\oob_monitor.py", line 52, in oob_percentage
    intersection = car_bbox_polygon.intersection(self.road_polygon.right_polygon)
  File "D:\tool-competition-av\.venv-36\lib\site-packages\shapely\geometry\base.py", line 620, in intersection
    return geom_factory(self.impl['intersection'](self, other))
  File "D:\tool-competition-av\.venv-36\lib\site-packages\shapely\topology.py", line 70, in __call__
    self._check_topology(err, this, other)
  File "D:\tool-competition-av\.venv-36\lib\site-packages\shapely\topology.py", line 38, in _check_topology
    self.fn.__name__, repr(geom)))
shapely.errors.TopologicalError: The operation 'GEOSIntersection_r' could not be performed. Likely cause is invalidity of the geometry <shapely.geometry.polygon.Polygon object at 0x000001E17ED27978>
D:\tool-competition-av\simulations\beamng_executor\sim_2020-11-23--14-21-29

**Comment by alessiogambi** [comment]
There's some logic that repeats the experiment a number of times before giving up. I set this to try one or two times tops.
We still need to report this as failed experiment, suggesting that the test might have been invalid

**Comment by alessiogambi** [comment]
Fixed in ac4fd7f1837906c5d4bc3a8d6e6b04decf0fc15d
